src/System.Net.Http/tests/UnitTests/HttpProxy.cs

```diff
@@ -0,0 +1,157 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Net.Http;
+using Xunit;
+using Xunit.Abstractions;
+using System.Diagnostics;
+
+namespace System.Net.Http.Tests
+{
+    public class HttpEnvironmentProxyTest : RemoteExecutorTestBase
+    {
+        private readonly ITestOutputHelper _output;
+        private static readonly Uri fooHttp = new Uri("http://foo.com");
+        private static readonly Uri fooHttps = new Uri("https://foo.com");
+
+        // This will clean specific environmental variables
+        // to be sure they do not interfere with the test.
+        private void cleanEnv()
+        {
+            List<string>  vars = new List<string>() { "http_proxy", "HTTPS_PROY", "https_proxy",
+                                                      "all_proxy", "ALL_PROXY", 
+                                                      "NO_PROXY" };
+            foreach (string v in vars)
+            {
+                Environment.SetEnvironmentVariable(v, null);
+            }
+        }
+
+        public HttpEnvironmentProxyTest(ITestOutputHelper output)
+        {
+            _output = output;
+            cleanEnv();
+        }
+
+        [Fact]
+        public void getBasicProxy()
+        {
+            RemoteInvoke(() =>
+            {
+
+                IWebProxy p;
+                Uri u;
+
+                // It should not return object if there are no variables set.
+                Assert.True(HttpEnvironmentProxy.TryToCreate() == null);
+
+                Environment.SetEnvironmentVariable("all_proxy", "http://1.1.1.1:3000");
+                p = HttpEnvironmentProxy.TryToCreate();
+                Assert.True(p != null);
+                Assert.True(p.Credentials == null);
+
+                u = p.GetProxy(fooHttp);
+                Assert.True(u != null && u.Host == "1.1.1.1");
+                u = p.GetProxy(fooHttps);
+                Assert.True(u != null && u.Host == "1.1.1.1");
+
+                Environment.SetEnvironmentVariable("http_proxy", "http://1.1.1.2:3001");
+                p = HttpEnvironmentProxy.TryToCreate();
+                Assert.True(p != null);
```

Comment:
Instead of using this pattern to test for NULL or not-Null, you should use "Assert.Null" or "Assert.NotNull".